src/auth0-sso-login.js

```diff
@@ -122,87 +134,39 @@ export default class auth {
    * @description Ensure user is logged in:
    * 1. Check if there is an existing, valid token.
    * 2. Try logging in using an existing SSO session.
-   * 3. If auth0 lock widget is not explicitly disabled, try logging in using auth0 lock.
+   * 3. If universal login is not explicitly disabled, try logging in via the hosted login
    *
    * @param {Object}     configuration object
-   * @param {Boolean}    configuration.enableLockWidget whether auth0 lock should open when SSO
+   * @param {Boolean}    configuration.enabledHostedLogin whether the universal login should open when SSO
    *                     session is invalid; default = true
    * @param {Boolean}    configuration.forceTokenRefresh if token should be refreshed even if it may
    *                     be still valid; default = false
    * @return {Promise<>} empty resolved promise after successful login; rejected promise with error
    *                     otherwise
    */
-  ensureLoggedIn(configuration = { enableLockWidget: true, forceTokenRefresh: false }) {
+  ensureLoggedIn(configuration = { enabledHostedLogin: true, forceTokenRefresh: false }) {
     // if there is still a valid token, there is no need to initiate the login process
-    const latestAuthResult = this.getLatestAuthResult();
+    const latestAuthResult = this.getIdToken();
     if (!configuration.forceTokenRefresh && latestAuthResult &&
       this.tokenExpiryManager.getRemainingMillisToTokenExpiry() > 0) {
       return Promise.resolve();
     }
 
-    let options = {
-      auth: {
-        params: {
-          responseType: 'id_token token',
-        },
-        redirect: false,
-      },
-      closable: false,
-    };
-    if (this.config.auth0LockOptions) {
-      options = Object.assign(options, this.config.auth0LockOptions);
-    }
+    this.redirectHandler.attemptRedirect();
 
     const authPromise = this.renewAuthSequencePromise
       .then(() => this.renewAuth())
       .catch((e) => {
-        // if auth0 lock is not enabled, error out
-        if (!configuration.enableLockWidget) {
+        // if universal login is not enabled, error out
+        if (!configuration.enabledHostedLogin) {
```

Comment:
I wonder whether we even want this option. I can't really find a use case. (Unsure why we added it in the original version.)

Reply:
We added it because if you are on a screen which does not require login you should still show the user if they are logged in.  I.e. a contact or documentation screen.  It doesn't make sense to force them to login, but it also makes sense to let them stil llogout.